UsbIpServer/ConnectedClient.cs

```diff
@@ -130,16 +130,21 @@ async Task HandleRequestImportAsync(CancellationToken cancellationToken)
                     using var attachedClientTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
                     Watcher.WatchDevice(busid, () => attachedClientTokenSource.Cancel());
                     var attachedClientToken = attachedClientTokenSource.Token;
+                    RegistryUtils.SetDeviceAsAttached(exportedDevice);
+                    var iPEndPoint = ClientContext.TcpClient.Client.RemoteEndPoint as IPEndPoint;
+                    RegistryUtils.SetDeviceAddress(exportedDevice, iPEndPoint!.Address.MapToIPv4().ToString());
```

Comment:
usbipd works perfectly fine with IPv6. If you just want a more readable address than [::ffff:a.b.c.d] (an IPv6 representation of an IPv4 address, due to the single dual-stack socket) then make a helper that first checks if the address is in fact IPv4 before doing the MapToIPv4.

Reply:
fixed